omni-relayer/src/startup/evm.rs

```diff
@@ -94,238 +58,64 @@ pub async fn start_indexer(
-            warn!(
-                "Failed to parse token as AccountId: {:?}",
-                init_log.inner.token
-            );
-            return;
-        };
-
-        let Ok(recipient) = init_log.inner.recipient.parse::<AccountId>() else {
-            warn!(
-                "Failed to parse recipient as AccountId: {:?}",
-                init_log.inner.recipient
-            );
-            return;
-        };
-
-        let sender = config.near.token_locker_id.clone();
+    let tx_hash = H256::from_slice(tx_hash.as_slice());
 
-        // If storage is sufficient, then flag should be false, otherwise true
-        let sender_is_storage_deposit =
-            !utils::storage::is_storage_sufficient(jsonrpc_client, &token, &sender)
-                .await
-                .unwrap_or_default();
-        let recipient_is_storage_deposit =
-            !utils::storage::is_storage_sufficient(jsonrpc_client, &token, &recipient)
-                .await
-                .unwrap_or_default();
-
-        let fin_transfer_args = FinTransferArgs {
-            chain_kind: ChainKind::Eth,
-            native_fee_recipient: config.evm.relayer_address_on_eth.clone(),
-            storage_deposit_args: StorageDepositArgs {
-                token,
-                accounts: vec![
-                    (sender, sender_is_storage_deposit),
-                    (recipient, recipient_is_storage_deposit),
-                ],
-            },
-            prover_args,
-        };
-
-        let mut serialized_fin_transfer_args = Vec::new();
-        if let Err(err) = fin_transfer_args.serialize(&mut serialized_fin_transfer_args) {
-            warn!("Failed to serialize fin transfer args: {}", err);
-            return;
-        }
+    let Some(block_number) = log.block_number else {
+        warn!("No block number in log: {:?}", log);
+        return;
+    };
 
+    if log.log_decode::<utils::evm::InitTransfer>().is_ok() {
         utils::redis::add_event(
             redis_connection,
             utils::redis::ETH_WITHDRAW_EVENTS,
             tx_hash.to_string(),
-            serialized_fin_transfer_args,
+            (block_number, log, tx_logs),
         )
         .await;
-    } else if log.log_decode::<FinTransfer>().is_ok() {
-        let claim_fee_args = ClaimFeeArgs {
-            chain_kind: ChainKind::Eth,
-            prover_args,
-            native_fee_recipient: config.evm.relayer_address_on_eth.clone(),
-        };
-
-        let mut serialized_claim_fee_args = Vec::new();
-        if let Err(err) = claim_fee_args.serialize(&mut serialized_claim_fee_args) {
-            warn!("Failed to serialize claim fee args: {}", err);
-            return;
-        }
-
+    } else if log.log_decode::<utils::evm::FinTransfer>().is_ok() {
         utils::redis::add_event(
             redis_connection,
             utils::redis::FINALIZED_TRANSFERS,
             tx_hash.to_string(),
-            serialized_claim_fee_args,
+            log,
```

Comment:
Why we don't save the block number here as well?

Reply:
In case of FinTransferEvent, before we are doing the "cliam_fee", we should check the bock number as well.

Reply:
Fixed in 998b21e